src/main/scala/fs2/async/mutable/Queue.scala

```diff
@@ -0,0 +1,109 @@
+package fs2.async.mutable
+
+import fs2._
+
+import fs2.async.immutable
+
+
+/**
+ * Asynchronous queue interface. Operations are all nonblocking in their
+ * implementations, but may be 'semantically' blocking. For instance,
+ * a queue may have a bound on its size, in which case enqueuing may
+ * block until there is an offsetting dequeue.
+ */
+trait Queue[F[_],A] {
+
+  /**
+   * A `Sink` for enqueueing values to this `Queue`.
+   */
+  //todo: how about sinks?
+  //def enqueue: Sink[Task, A]
+
+  /**
+   * Enqueue one element in this `Queue`. Resulting task will
+   * terminate with failure if queue is closed or failed.
+   * Please note this will get completed _after_ `a` has been successfully enqueued.
+   * @param a `A` to enqueue
+   */
+  def enqueueOne(a: A): F[Unit]
+
+  /**
+   * Enqueue multiple `A` values in this queue. This has same semantics as sequencing
+   * repeated calls to `enqueueOne`.
+   */
+  def enqueueAll(xa: Seq[A]): F[Unit]
+
+  /**
+   * Provides a process that dequeue from this queue.
+   * When multiple consumers dequeue from this queue,
+   * they dequeue in first-come, first-serve order.
+   *
+   * Please use `Topic` instead of `Queue` when all subscribers
+   * need to see each value enqueued.
+   *
+   * This process is equivalent to `dequeueBatch(1)`.
+   */
+  def dequeue: Stream[F, A]
+
+  /**
+   * Provides a process that dequeues in chunks.  Whenever *n* elements
+   * are available in the queue, `min(n, limit)` elements will be dequeud
+   * and produced as a single `Seq`.  Note that this naturally does not
+   * *guarantee* that `limit` items are returned in every chunk.  If only
+   * one element is available, that one element will be returned in its own
+   * sequence.  This method basically just allows a consumer to "catch up"
+   * to a rapidly filling queue in the case where some sort of batching logic
+   * is applicable.
+   */
+  // todo: likely we don't need it as we have Chunks now?
```

Comment:
Yeah, this is not needed. The consumer can await whole chunks and catch up that way. Let's kill it.